bika/lims/workflow/sample/events.py

```diff
@@ -122,23 +122,20 @@ def after_sample(obj):
     """
     _cascade_transition(obj, 'sample')
 
+    def next_transition(o):
+        return 'receive' if isTransitionAllowed(o, 'receive') else 'sample_due'
```

Comment:
This `next_transition` function is only called once. I think is not required and you could just do:
```python
low_state_action = 'sample_due'
if isTransitionAllowed(obj, 'receive'):
    low_state_action = 'receive'
```